clang/lib/AST/MicrosoftMangle.cpp

```diff
@@ -1938,12 +1946,23 @@ void MicrosoftCXXNameMangler::mangleTemplateArgValue(QualType T,
         mangleNumber(V.getLValueOffset().getQuantity());
       } else if (!V.hasLValuePath()) {
         // FIXME: This can only happen as an extension. Invent a mangling.
-        break;
+        DiagnosticsEngine &Diags = Context.getDiags();
+        unsigned DiagID =
+            Diags.getCustomDiagID(DiagnosticsEngine::Error,
+                                  "cannot mangle this template argument yet "
+                                  "(non-null base with null lvalue path)");
```

Comment:
I think that "lvalue base" and "lvalue path" are clang internal terms. Not sure whether they are worth appearing in diagnostics.

Reply:
I just want something to differentiate the 5 otherwise identical errors to save some future user that step when reporting or fixing one of them. I'm open for suggests on phrasing, anything other than going back to the same message in all 5 cases.